File: melosys-eessi-app/src/main/java/no/nav/melosys/eessi/controller/dto/SaksrelasjonDto.java
package no.nav.melosys.eessi.controller.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import no.nav.melosys.eessi.models.FagsakRinasakKobling;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class SaksrelasjonDto {
    private Long gsakSaksnummer;
    private String rinaSaksnummer;
    private String bucType;

    public static SaksrelasjonDto av(FagsakRinasakKobling fagsakRinasakKobling) {
        return new SaksrelasjonDto(
                fagsakRinasakKobling.getGsakSaksnummer(),
                fagsakRinasakKobling.getRinaSaksnummer(),
                fagsakRinasakKobling.getBucType().name()
        );
    }
}

File: melosys-eessi-app/src/main/java/no/nav/melosys/eessi/controller/dto/SaksrelasjonDto.kt
package no.nav.melosys.eessi.controller.dto

import no.nav.melosys.eessi.models.FagsakRinasakKobling

data class SaksrelasjonDto(
    var gsakSaksnummer: Long? = null,
    var rinaSaksnummer: String? = null,
    var bucType: String? = null
) {
    companion object {
        @JvmStatic
        fun av(fagsakRinasakKobling: FagsakRinasakKobling): SaksrelasjonDto =
            SaksrelasjonDto(
                gsakSaksnummer = fagsakRinasakKobling.gsakSaksnummer,
                rinaSaksnummer = fagsakRinasakKobling.rinaSaksnummer,
                bucType = fagsakRinasakKobling.bucType.name
            )
    }
}
